src/rockstor/storageadmin/static/storageadmin/js/router.js

```diff
@@ -757,7 +758,30 @@ var AppRouter = Backbone.Router.extend({
         this.currentLayout = new LogsView();
         $('#maincontent').empty();
         $('#maincontent').append(this.currentLayout.render().el);
-    }
+    },
+	
+	showShell: function() {
+		//Special router function for shell
+		//We fetch shellinaboxd service model and check for detach option
+		//If not present or false we go with normal layout to maincontent
+		//otherwhise (popup enabled) we open it in a new detached window
+		var _this = this;
+		_this.shell_service = new Service({name: 'shellinaboxd'});
+		_this.shell_service.fetch({
+		success: function(collection){
+			var config = JSON.parse(collection.get('config'));
+			if ('detach' in config && config.detach){
```

Comment:
Default config got via prep_db on 852d737 and 41ca9cd

There's also a new dictionary (services_configs) everyone can use to set a default config field val on services table :)

Flyer